components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -61,6 +65,46 @@ interface NimbusApi : Observable<NimbusApi.Observer> {
      */
     fun updateExperiments() = Unit
 
+    /**
+     * Fetches experiments from the RemoteSettings server.
+     *
+     * This is performed on a background thread.
+     *
+     * Notifies `onExperimentsFetched` to observers once the experiments has been fetched from the
+     * server.
+     *
+     * Notes:
+     * * this does not affect experiment enrolment, until `applyPendingExperiments` is called.
+     * * this will overwrite pending experiments previously fetched with this method, or set with
+     *   `setExperimentsLocally`.
+     */
+    fun fetchExperiments() = Unit
+
+    /**
+     * Calculates the experiment enrolment from experiments from the last `fetchExperiments` or
+     * `setExperimentsLocally`, and then informs Glean of new experiment enrolment.
+     *
+     * Notifies `onUpdatesApplied` once enrolments are recalculated.
+     */
+    @MainThread
+    fun applyPendingExperiments() = Unit
```

Comment:
Does this have the potential to do some I/O as it applies the updates in the database in nimbus-sdk?